### PR TITLE
Loss Ops: Adds Softmax Cross Entropy Loss

### PR DESCRIPTION
**Pull request by ManrajGrover**
#### Description
This PR adds `tf.losses.softmaxCrossEntropy` with weights support.

1. Renames current `tf.losses.softmaxCrossEntropy` -> `tf.softmaxCrossEntropyWithLogits` (since loss function depends on it, can be moved to loss function itself)
2. Adds support for `tf.losses.softmaxCrossEntropy` with weighted loss.

---

##### For repository owners only:

Please remember to apply all applicable tags to your pull request.
Tags: FEATURE, BREAKING, BUG, PERF, DEV, DOC, SECURITY

For more info see: https://github.com/tensorflow/tfjs/blob/master/DEVELOPMENT.md

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/1133)

### CODE REVIEW

**Comment by ManrajGrover** [comment]
Ah, not sure what to do here. Since softmax makes use of `log`, for Safari any value less than `1e-4` gets converted to `0` which `log` returns as `NaN`. And if values are clipped at `1e-4`, test cases fail.

**Comment by ManrajGrover** [comment]
@dsmilkov Made the changes
